app/src/main/java/com/eis0/library_demo/ui/main/PlaceholderFragment.java

```diff
@@ -0,0 +1,94 @@
+package com.eis0.library_demo.ui.main;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ListAdapter;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.ListFragment;
+
+import com.eis0.library_demo.DataProvider;
+
+import java.util.Observer;
+
+/**
+ * A placeholder fragment containing one of the three list types depending on the section number.
+ *
+ * @author Matteo Carnelos
+ */
+public class PlaceholderFragment extends ListFragment {
+
+    private static final String ARG_SECTION_NUMBER = "section_number";
+    private static final int INCOMING_SECTION_NUMBER = 1;
+    private static final int OPENED_SECTION_NUMBER = 2;
+    private static final int CLOSED_SECTION_NUMBER = 3;
+    private DataProvider dataProvider = DataProvider.getInstance();
+
+    /**
+     * Returns a new PlaceholderFragment instance with the bundle containing the section number
+     * specified in the index parameter.
+     *
+     * @param index The index of the PlaceholderFragment section.
+     * @return A PlaceholderFragment instance with the bundle containing the section number.
+     * @author Matteo Carnelos
+     */
+    static PlaceholderFragment newInstance(int index) {
```

Comment:
If this method is intended as the only way of properly instantiating this class, then it should have the same visibility as the class itself or vice-versa, otherwise default constructor might be called from outside.

Reply:
Actually this method is unreachable from classes outside the package (it has default package-private access modifier, that is the strictest possible in this case) and the `PlaceholderFragment` class cannot be non-public, because it extends ListFragment. And so, as they say:

> PlaceholderFragment must be a public static class to be  properly recreated from instance state.

Reply:
Ok, so newInstance is not the only way to instantiate the class because the constructor has to be called from outside, then it's totally fine the way it is.